src/front/glsl/builtins.rs

```diff
@@ -1670,12 +1670,19 @@ fn texture_call(
     meta: SourceMetadata,
 ) -> Result<Handle<Expression>> {
     if let Some(sampler) = ctx.samplers.get(&image).copied() {
+        let mut array_index = comps.array_index;
+
+        if let Some(mut array_index_expr) = array_index {
```

Comment:
You can use `ref mut` here and remove the reference afterwards and the assignment

Reply:
Thanks, that helps clean it up a lot.